src/custom/prelude.rs

```diff
@@ -0,0 +1,230 @@
+use std::{any::TypeId, collections::HashSet};
+
+use anyhow::{anyhow, Result};
+use hugr::{
+    extension::prelude::{self, ConstUsize, QB_T, USIZE_T},
+    ops::constant::CustomConst,
+    types::TypeEnum,
+    HugrView,
+};
+use inkwell::{
+    types::{BasicType, BasicTypeEnum, IntType},
+    values::BasicValueEnum,
+};
+
+use crate::{emit::func::EmitFuncContext, types::TypingSession};
+
+use super::{CodegenExtension, CodegenExtsMap};
+
+/// A helper trait for implementing [CodegenExtension]s for
+/// [hugr::extension::prelude].
+///
+/// All methods have sensible defaults provided, and [DefaultPreludeCodegen] is
+/// trivial implementation o this trait, which delegates everything to those
+/// default implementations.
+///
+/// One should use either [PreludeCodegenExtension::new], or
+/// [CodegenExtsMap::add_prelude_extensions] to work with the
```

Comment:
So, if for some reason we wanted to tweak code generation for prelude constructs for different targets, we have the flexibility of defining code generation for this new target by implementing the PreludeCodegen trait? That seems very useful if we want to create custom lowering to QIR for other targets that is spec compliant.